### PR TITLE
Updating the README

### PR DESCRIPTION
**Pull request by SimonRichardson**
The README was way out of date, but also adding information about the
basics of a charm in terms of the configuration options. This doesn't go
into the differences between a bash charm, nor a reactive or operator
charm, just what a charm is in terms of configration.

### CODE REVIEW

**Review by achilleasa — APPROVED** [review]
~You should also land this against the V2 branch (its HEAD is the last PR you merged before my recent changes)~ Ignore comment; I was referring to another repo.

**Comment by SimonRichardson** [comment]
$$merge$$

File: README.md [hunk 1]
```diff
@@ -3,9 +3,93 @@ Juju charms
...
-Stable versions of this API are available on gopkg.in at
-gopkg.in/juju/charm.vD where D is a version spec.  If you are viewing this
-readme on github.com you can click the 'branch:' button above to view tags
-and branches.  See http://labix.org/gopkg.in for more information.
+A bare minimum charm consists of a directory containing a series of files and
+directories that inform Juju how to correctly handle and execute a charm.
+
+The [charming docs](https://discourse.juju.is/c/docs/charming-docs) provide more
+advanced information and tutorials.
+The following is a very simple basic configuration guide.
```
Comment (manadart) [inline README.md:13]: "simple basic" is repetitive.

File: README.md [hunk 2]
```diff
@@ -3,9 +3,93 @@ Juju charms
...
+
+### `metadata.yaml`
+
+The `metadata.yaml` is a required charm configuration yaml. It is expected that
+every charm contains a `metadata.yaml`.
+
+An example of a very basic `metadata.yaml`:
```
Comment (manadart) [inline README.md:20]: I use "minimal" here.

**Pull request merged**